test/modules/TestResultCoordinator/Services/TestResultReportingService.cs

```diff
@@ -39,16 +40,19 @@ public TestResultReportingService(ITestOperationResultStorage storage)
                             connectivitySettings.TestVerificationDelay;
                         // Set sendReportFrequency to -1ms to indicate that the sending report Timer won't repeat
                         this.sendReportFrequency = TimeSpan.FromMilliseconds(-1);
+                        this.logUploadDuration = Option.None<TimeSpan>();
                         break;
                     }
 
                 case TestMode.LongHaul:
                     {
+                        // In long haul mode, wait 1 report frequency before starting
+                        this.delayBeforeWork = this.sendReportFrequency;
```

Comment:
Why did you move this? You're reading the value of `this.sendReportFrequency` _before_ you set it (the next line).

Reply:
Wanted to make the assignment order consistent with above block. Got confused as `Settings.Current` has its own `SendReportFrequency`. Changed it back.

Will request review when both connectivity and longhaul are passing.